bl/validation/k8sValidator.go

```diff
@@ -56,7 +60,7 @@ func (val *K8sValidator) ValidateResources(filesConfigurationsChan chan *extract
 			}
 		}
 	}()
-	return validK8sFilesConfigurationsChan, invalidK8sFilesChan
+	return validK8sFilesConfigurationsChan, invalidK8sFilesChan, k8sValidationWarningPerValidFile
```

Comment:
You return this value before it's being written to, so potentially you can read it outside the function before it's being populated, either return this value in a channel or initialize it in the same goroutine that returns the value